src/integration/resources/inprocess/dbnode.go

```diff
@@ -164,24 +166,38 @@ func (d *dbNode) start() {
 	d.started = true
 }
 
-func (d *dbNode) HostDetails(port int) (*admin.Host, error) {
-	// TODO(nate): implement once working on helpers for spinning up
-	// a multi-node cluster since that's what it's currently being
-	// used for based on the docker-based implementation.
-	// NOTES:
-	// - id we can generate
-	// - isolation_group set a constant?
-	// - weight just use 1024?
-	// - address is 0.0.0.0
-	// - port is the listen address (get from config)
+func (d *dbNode) HostDetails(_ int) (*admin.Host, error) {
+	_, p, err := net.SplitHostPort(d.cfg.DB.ListenAddressOrDefault())
+	if err != nil {
+		return nil, err
+	}
+
+	port, err := strconv.Atoi(p)
+	if err != nil {
+		return nil, err
+	}
+
+	hostID, err := d.cfg.DB.HostIDOrDefault().Resolve()
+	if err != nil {
+		return nil, err
+	}
+
+	discoverCfg := d.cfg.DB.DiscoveryOrDefault()
+	envConfig, err := discoverCfg.EnvironmentConfig(hostID)
+	if err != nil {
+		return nil, err
+	}
 
 	return &admin.Host{
```

Comment:
IsolationGroup is empty? Will that affect the sharding algorithm?

Reply:
Ah yeah, I've got a change to support in an upcoming PR. For now, it's OK to be left blank.